### ISSUE TITLE
0.14.0 breaks generated YAML

### ISSUE THREAD

**Issue by burner-account** [nats-io/k8s#466]
Relocating the data part of [helm/charts/nats/templates/configmap.yaml](https://github.com/nats-io/k8s/commit/3e259f1980fe5db9b369f3465e0e1e51ad316047#diff-a135d042cfafb1f4f5f7ce8b1bbb4aefba0e2b793035c3b09a55b74cc8663175) to [helm/charts/nats/files/configmap/nats.conf](https://github.com/nats-io/k8s/commit/3e259f1980fe5db9b369f3465e0e1e51ad316047#diff-b6b6aa34a014b3ddd8915cd5c332e0bfdef655505c762957753625266dff643d) seems to have introduced an  error (formatting/line-break).

If you `helm template` nats/nats, you get
```
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: nats-core-config
  namespace: nats-ns
  labels:
    helm.sh/chart: nats-0.14.0
    app.kubernetes.io/name: nats
    app.kubernetes.io/instance: nats-core
    app.kubernetes.io/version: "2.7.3"
    app.kubernetes.io/managed-by: Helm
data:
  nats.conf: |
    <... snip ...>
    #####################
    #                   #
    # TLS Configuration #
    #                   #
    #####################tls {
      cert_file: /etc/nats-certs/clients/nats-server-tls/tls.crt
      key_file:  /etc/nats-certs/clients/nats-server-tls/tls.key
      ca_file: /etc/nats-certs/clients/nats-server-tls/ca.crt
  }

```
This seems to have some trouble with the "tls" subtree.

I think this is the reason why i get `Error: INSTALLATION FAILED: YAML parse error on nats/templates/configmap.yaml: error converting YAML to JSON: yaml: line 34: did not find expected key"` returned by helm install.

**Comment by burner-account** [comment]
If this report can be confirmed, the logical follow-up would be to massage some YML validation into the CI pipeline?
Such errors are preventable.

**Comment by caleblloyd** [comment]
Will take a look

> the logical follow-up would be to massage some YML validation into the CI pipeline

The YAML validation currently only runs on the default `values.yaml` file, we probably also need a "fully loaded" `values.yaml` file where every single option is set, so that every single option gets validated.

**Comment by burner-account** [comment]
@caleblloyd  Thank you very much! I appreciate your explanation regarding the validation.

**Comment by caleblloyd** [comment]
I think that the issue is with `{{- tpl (.Files.Glob "files/configmap/*").AsConfig . | nindent 2 }}` - that is doing the templating after some indentation has already been added.  So the templating engine has the capability to remove this indentation when it encounters `{{-` 

I'm going to attempt to back that particular change out and revert back to the configmap until we have some better config checking in-place.  I think that we can still sha256sum the configmap so it should be OK

**Comment by caleblloyd** [comment]
Good catch.  But I am not sure what else may have been broken by it, so we'll just wholesale revert back to using the ConfigMap for now.

YAML templating is already painful enough, and introducing more indentations prior to templating with `Files.AsConfig` makes it too confusing.

We will probably try to move it back out to `.Files` again so that we can edit a `.conf` file and not a string block.  But we will need to do the templating prior to the indentation so that rules out `Files.AsConfig`.  And will need to validate that we don't introduce errors like the one in this issue.

**Comment by caleblloyd** [comment]
Should be fixed in [nats-0.14.1](https://github.com/nats-io/k8s/releases/tag/nats-0.14.1)
